React/CoreModules/RCTActionSheetManager.mm

```diff
@@ -131,11 +138,23 @@ - (void)presentViewController:(UIViewController *)alertController
     }
 
     NSInteger localIndex = index;
-    [alertController addAction:[UIAlertAction actionWithTitle:option
-                                                        style:style
-                                                      handler:^(__unused UIAlertAction *action) {
-                                                        callback(@[ @(localIndex) ]);
-                                                      }]];
+    UIAlertAction *action = [UIAlertAction actionWithTitle:option
+                                                     style:style
+                                                   handler:^(__unused UIAlertAction *action) {
+                                                     callback(@[ @(localIndex) ]);
+                                                   }];
+      
+    if (index < icons.count) {
+      UIImage *iconImage = [self iconImageFromDictionary:icons[index]];
+      if (iconImage) {
+        if (!tintIcons) {
+          iconImage = [iconImage imageWithRenderingMode:UIImageRenderingModeAlwaysOriginal];
+        }
+        [action setValue:iconImage forKey:@"image"];
+      }
```

Comment:
i feel it is probably better to have an error here instead of failing silently, if the consumer is passing down a list of icons then presumably iconImage should never be nil

Reply:
This was origionally done to allow passing in a `null` value for some of the icons, as described above. However, I think you are correct that it is better to show an error when a value has been passed in.

I have changed the code so it will ignore `null` values, but will show an error when something has been passed in.